mtls/rotator.go

```diff
@@ -643,8 +643,11 @@ func (cr *CertRotator) generateNamespaceTLS(namespace string) (*corev1.Secret, e
 	artifactCache := cr.artifactCaches[n-1]
 	caArtifacts := artifactCache.ca
 
-	hostname := fmt.Sprintf("*.%s.pod.%s", namespace, cr.ClusterDomain)
-	cert, key, err := cr.createCertPEM(caArtifacts, hostname, time.Now().Add(-1*time.Hour), caArtifacts.validUntil)
+	hostnames := []string{
+		fmt.Sprintf("*.%s.pod.%s", namespace, cr.ClusterDomain),
+		fmt.Sprintf("*.tf-runner.%s.svc.%s", namespace, cr.ClusterDomain),
```

Comment:
Maybe we introduce `--use-pod-subdomain-resolution` to also guard here.

Reply:
Yeah, will do !

Reply:
Done